src/backend/optimizer/plan/planner.c

```diff
@@ -1866,6 +1906,352 @@ grouping_planner(PlannerInfo *root, double tuple_fraction)
 		 */
 		current_rel = query_planner(root, standard_qp_callback, &qp_extra);
 
+		/*
+		 * Answer Query Using Materialized Views(AQUMV).
+		 */
+		RelOptInfo *mv_final_rel = NULL;
+		bool can_not_use_mv = parse->hasAggs ||
+							  parse->hasWindowFuncs ||
+							  parse->hasDistinctOn ||
+							  parse->hasModifyingCTE ||
+							  parse->sortClause ||
+							  (parse->parentStmtType == PARENTSTMTTYPE_REFRESH_MATVIEW) ||
+							  (parse->parentStmtType == PARENTSTMTTYPE_CTAS) ||
+							  parse->hasSubLinks;
+
+		if (Gp_role == GP_ROLE_DISPATCH &&
+			answer_query_using_materialized_views &&
+			!can_not_use_mv &&
+			(list_length(parse->jointree->fromlist) == 1))
+		{
+			Node	   *jtnode = (Node *) linitial(parse->jointree->fromlist);
+
+			if (IsA(jtnode, RangeTblRef))
+			{
+				int			varno = ((RangeTblRef *) jtnode)->rtindex;
+				RangeTblEntry *rte = root->simple_rte_array[varno];
+
+				Assert(rte != NULL);
+
+				/*
+				 * AQUMV_FIXME_MVP:
+				 *	Single relation, excluding catalog.
+				 */
+				if (rte->rtekind == RTE_RELATION && !IsSystemClassByRelid(rte->relid))
+				{
+
+					Relation		ruleDesc;
+					Relation		matviewRel;
+					SysScanDesc		rcscan;
+					HeapTuple		tup;
+					RewriteRule		*rule;
+					Form_pg_rewrite	rewrite_tup;
+					List			*actions;
+					Query			*mvQuery;
+					Query			*mvRelQueryTree; /* Query tree of select from mv itself. */
+					Node			*mvjtnode;
+					bool			need_close = false;
+
+					ruleDesc = table_open(RewriteRelationId, AccessShareLock);
+
+					rcscan = systable_beginscan(ruleDesc, InvalidOid, false,
+												NULL, 0, NULL);
+
+					while (HeapTupleIsValid(tup = systable_getnext(rcscan)))
+					{
+						CHECK_FOR_INTERRUPTS();
+						if (need_close)
+							table_close(matviewRel, AccessShareLock);
+
+						rewrite_tup = (Form_pg_rewrite) GETSTRUCT(tup);
+
+						matviewRel = table_open(rewrite_tup->ev_class, AccessShareLock);
+						need_close = true;
+
+						/* AQUMV_FIXME_MVP: TOOD use Incremental Materialized View. */
+						if (matviewRel->rd_rel->relkind != RELKIND_MATVIEW)
+							continue;
+
+						rule = matviewRel->rd_rules->rules[0];
+
+						/* Filter a SELECT action, and not instead. */
+						if ((rule->event != CMD_SELECT) || !(rule->isInstead))
```

Comment:
need to consider `ev_enabled` ?

Reply:
I take a look at this param, in doc:
> ev_enabled char
Controls in which [session_replication_role](https://github.com/cloudberrydb/cloudberrydb/pull/runtime-config-client.html#GUC-SESSION-REPLICATION-ROLE) modes the rule fires. O = rule fires in “origin” and “local” modes, D = rule is disabled, R = rule fires in “replica” mode, A = rule fires always.

It's used for replication-related triggers and rules for the current session.

And by checking the codes, it's only used in pg_dump, I didn't find any other places using it with materialized views.
Shall we consider a mv as valid by `ev_enabled`?